Docsp-21583

ianf-mongodb opened the pull request.

jeff-allen-mongo commented on source/query/queries.txt:
```diff
@@ -6,6 +6,14 @@ View Past Queries
 .. default-domain:: mongodb
 
+Starting in |compass| 1.31, you can view your favorite and saved 
+aggregation queries on the "My Queries" view once connected to 
```
[minor] Suggest using a guilabel for "My Queries"

```suggestion
aggregation queries on the :guilabel:`My Queries` view once connected to 
```

ianf-mongodb replied:
Updated.

jeff-allen-mongo commented on source/query/queries.txt:
```diff
@@ -6,6 +6,14 @@ View Past Queries
 .. default-domain:: mongodb
 
+Starting in |compass| 1.31, you can view your favorite and saved 
+aggregation queries on the "My Queries" view once connected to 
```
Maybe "aggregation queries" should be "queries and aggregation pipelines"? To make it clear that this view will show both?

ianf-mongodb replied:
Good observation, I did not notice this until you mentioned it. I updated accordingly.

jeff-allen-mongo commented on source/query/queries.txt:
```diff
@@ -6,6 +6,14 @@ View Past Queries
+.. figure:: /images/compass/query-my-queries-splash.png
+   :figwidth: 720 px
+   :alt: My queries view
+
```
I'm slightly confused about the placement of this content. The title of the page is "View Past Queries", but it seems to discuss favorite queries. We have a sub-page under this one for [favorite queries](https://docs-mongodbcom-staging.corp.mongodb.com/compass/docsworker-xlarge/DOCSP-21583/query/favorites/). I feel like we should also be updating that page as part of these changes?

ianf-mongodb replied:
I spoke with Rhys previously to get a better idea of the ask for this ticket. This is the page he wanted the update on since we are trying to emphasize the new screen / experience of not having to go to the favorite/history view.

jeff-allen-mongo replied:
So then maybe this page title should be updated to something like "View Favorite Queries" or something more general? The title doesn't really seem to match the content.

jeff-allen-mongo reviewed (changes requested):
@ianf-mongodb thanks for the updates. See comments for some comments / questions.

ianf-mongodb reviewed (commented):
back to you @jeff-allen-mongo.

jeff-allen-mongo commented on source/query/queries.txt:
```diff
@@ -6,6 +6,14 @@ View Past Queries
 
+Starting in |compass| 1.31, you can view your favorite and saved 
+queries and aggregation pipelines on the :guilabel:`My Queries` view 
+once connected to your cluster.
```
Can we explain what happens when a user clicks a query or aggregation?

ianf-mongodb replied:
Added two small blurbs to address this.

ianf-mongodb reviewed (commented):
@jeff-allen-mongo updated page header in addition to your other suggestion.

jeff-allen-mongo commented:
Thanks @ianf-mongodb , I think the new page is clearer now, but I have a couple thoughts / questions. I think it might be easier to discuss these on Zoom, but I'm just leaving them here so we have them for the meeting.

- Perhaps it should be titled something like "View Saved Queries and Aggregations" because the new UX is all about saved queries and aggregations, rather than the query history.
- Maybe we should use headers on the updated page to delineate Saved Queries vs Past Queries?
- Do we still need the sub-page about loading favorite queries? That seems to be the old way of doing things, and it's a little confusing to have a sub-page describing the same task just done in a different way. I think the "Recent Queries" page still makes sense because that aspect of the experience didn't change.
- I think this new landing page needs a brief explanation of how to save favorite queries and pipelines. Maybe these could be links like, "To learn how to save a favorite query, see (link)".
- I think the Open and Save Save Aggregation Pipeline sections should be updated, letting users know that the pipeline will be saved in this new view: https://docs-mongodbcom-staging.corp.mongodb.com/compass/docsworker-xlarge/DOCSP-21583/aggregation-pipeline-builder/#open-a-saved-pipeline

ianf-mongodb commented:
@jeff-allen-mongo I have changes for what we discussed this morning.

A few other caveats 

- I renamed the page to "Managing Saved Queries and Aggregations" since we are discussing not only saving but also viewing the saved/favorites queries and pipelines. 

-  Added a few screen shots to the fact-save-pipeline.rst include, this is because it looked odd with all the other sections on the page which had screenshots.

- Added a redirect to cover any 404 risk from deleting source/query/favorites.txt

- Rearranged the ``past-query-view`` and ``favorite-queries`` refs that were displaced from this PR.

jeff-allen-mongo commented on source/aggregation-pipeline-builder.txt:
```diff
@@ -92,6 +80,14 @@ Open a Saved Pipeline
 
+.. note::
+
+   Starting in |compass| 1.31, you can view your favorite and saved 
+   queries and aggregation pipelines on the :guilabel:`My Queries` view 
```
Maybe omit "favorite" here and just say "saved queries and aggregation pipelines". I don't think users can "favorite" a pipeline so just generalizing to "saved queries and pipelines" is fine here.

jeff-allen-mongo commented on source/includes/fact-favorite-query.rst:
```diff
@@ -0,0 +1,26 @@
+You can favorite a query so that you can find it later. If you
+load a favorited query, you can change it without
+changing the original.
```
Maybe replace "original" with "saved version" ?

EDIT: I see you used "original saved copy" in source/includes/fact-save-pipeline.rst. Let's use that here as well.

jeff-allen-mongo commented on source/query/queries.txt:
```diff
@@ -1,25 +1,59 @@
+Managing Saved Queries and Aggregations
+=======================================
 
 .. default-domain:: mongodb
```
Can we include a toc on this page? Currently there's no right-hand toc.

jeff-allen-mongo commented on source/query/queries.txt:
```diff
@@ -1,25 +1,59 @@
 
-Compass automatically stores up to 20 most recent queries for each
-collection. To view these queries, click the history icon in the query bar.
+Saving an Aggregation Pipeline
```
It seems a bit weird to jump right from the title to a header. Can we add some sort of brief intro like:

Starting in Compass <version>, you can load saved queries and aggregations from the :guilabel:`My Queries` view once you are connected to your cluster. This page explains how to add and view favorite queries and aggregations."

jeff-allen-mongo commented on source/query/queries.txt:
```diff
@@ -1,25 +1,59 @@
+Viewing Favorited Queries
+-------------------------
+
+1. Click the :guilabel:`History` button at the top of the 
```
This might render a bit more nicely if we use the stepped procedure rst.

jeff-allen-mongo commented on source/includes/fact-favorite-query.rst:
```diff
@@ -0,0 +1,26 @@
+You can favorite a query so that you can find it later. If you
+load a favorited query, you can change it without
```
[Note for throughout]

Can we use "favorite" instead of "favorited"? "Favorited" doesn't sound quite right to me.

```suggestion
load a favorite query, you can change it without
```

jeff-allen-mongo commented on source/query/queries.txt:
```diff
@@ -1,25 +1,59 @@
+
+.. _saved-query-view:
+
+Using the :guilabel:`My Queries` View
```
[suggestion]

I think the content in this section should be elevated to the "Viewing Favorite Queries" section because this is the newly preferred method for viewing queries. Then, we can create a sub-section called "Using the :guilabel:`Favorites` Tab" that has the content that used to live here:

https://www.mongodb.com/docs/compass/current/query/favorites/.

So I think ultimately we want something like:

```
Viewing Favorited Queries
------------------------

Starting in  |compass| 1.31, you can view your favorited and saved 
queries and aggregation pipelines on the :guilabel:`My Queries` view.....

Using the :guilabel:`Favorites` Tab
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

You can also view favorite queries from the :guilabel:`Favorites` tab from in the :guilabel:`Documents` view. To open the :guilabel:`Favorites` tab:

1. Click the History button at the top of the Documents view...
2. ....
```

jeff-allen-mongo commented on source/query/queries.txt:
```diff
@@ -1,25 +1,59 @@
+
+.. _favorite-queries:
+
+Adding a Favorite Query
```
```suggestion
Saving a Favorite Query
```

jeff-allen-mongo commented on source/includes/fact-favorite-query.rst:
```diff
@@ -0,0 +1,26 @@
+
+To add a query to your favorites:
+
+1. Click the :guilabel:`History` button at the top of the pipeline
```
Maybe try the stepped procedure RST here as well?

ianf-mongodb replied:
Updated.

jeff-allen-mongo reviewed (commented):
@ianf-mongodb thanks for these edits. I think this new structure is much improved. I've left a couple more suggestions for you but I think this is much closer to where we want this to be.

ianf-mongodb reviewed (commented):
@jeff-allen-mongo this is ready for review, I lost your comments around converting the includes to step files, but I did make that change as well.

jeff-allen-mongo commented on source/includes/fact-save-pipeline.rst:
```diff
@@ -0,0 +1,21 @@
+
+To save your pipeline:
+
+1. Click the :guilabel:`Save` dropdown button at the top of the pipeline
```
Can we use `.. procedure::` here as well?

ianf-mongodb replied:
Updated.

jeff-allen-mongo commented on source/includes/steps-saving-favorite-queries.yaml:
```diff
@@ -0,0 +1,38 @@
+     :figwidth: 316 px
+     :alt: Save favorite query
+---
+title: Save
+level: 4
+ref: save-your-query
+content: |
+  Click :guilabel:`Save`.
```
For something like this we can omit the `content` and keep the action in the title. We can avoid the redundancy.

jeff-allen-mongo commented on source/query/queries.txt:
```diff
@@ -1,25 +1,69 @@
-   :alt: Query history view
+   :alt: My queries view
+
+You can also view favorite queries from the :guilabel:`Favorites` tab 
```
This should be in its own sub-section

ianf-mongodb replied:
Updated.

jeff-allen-mongo replied:
I don't see this as updated. We want a sub-section titled something like "Using the FavoritesTab" for this content"

ianf-mongodb reviewed (commented):
@jeff-allen-mongo Converted the steps -> procedures and added the subsection as suggested.

jeff-allen-mongo commented on source/includes/fact-save-pipeline.rst:
```diff
@@ -0,0 +1,26 @@
+.. procedure::
+   :style: normal
+
+   .. step:: Open query history
```
I don't think this step title is correct.

jeff-allen-mongo reviewed (approved):
LGTM % fixes

jeff-allen-mongo commented on source/query/queries.txt:
```diff
@@ -1,25 +1,73 @@
 
-   To learn how to write queries to filter your data, see
-   :ref:`query-bar`.
+For details on how to view query history see: 
```
```suggestion
For details on how to view query history see
```

Anemy reviewed (approved):
lgtm! Nice.
